library/common/network/configurator.h

```diff
@@ -85,5 +85,22 @@ class ConfiguratorHandle {
   Server::Configuration::FactoryContextBase& context_;
 };
 
+/**
+ * Provides nullable access to the singleton Configurator.
+ */
+class NullableConfiguratorHandle {
```

Comment:
Created this rather than having naked access via the singleton manager. Lmk what you think.

Reply:
Sure. Nit: I'd call this one the `ConfiguratorHandle` and the other the `ConfiguratorFactory`. (The reason the other is called "Handle" today is because I actually used a similar non-constructing mechanism for it at one point.)

Reply:
Updated!